File: java/AtbashCipher.java
import java.lang.StringBuilder;
import java.util.Scanner;

class AtbashCipher {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);

    System.out.println("Enter text to encode/decode:");
    String text = scanner.nextLine();

    text = text.toLowerCase();

    String result = atbashCipher(text);

    System.out.printf("Result: %s", result);

    scanner.close();
  }

  static String atbashCipher(String plainText) {
    StringBuilder sb = new StringBuilder();
    char c = 0;

    for (int i = 0; i < plainText.length(); i++) {
      char chr = plainText.charAt(i);

      if (chr >= 97 && chr <= 122) {
        c = (char)((25 - (chr - 97)) % 26 + 97);

        sb.append(c);
      } else {
        sb.append(chr);
      }
    }

    return sb.toString();
  }
}

File: javascript/atbash-cipher.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function input(question) {
  return new Promise((resolve) => {
    rl.question(question, resolve);
  });
}

function atbashCipher(text) {
  let result = "";
  let char = 0;

  for (let i = 0; i < text.length; i++) {
    const code = text.charCodeAt(i);

    if (code >= 97 && code <= 122) {
      char = (25 - (code - 97)) % 26 + 97;

      result += String.fromCharCode(char);
    } else {
      result += text[i];
    }
  }

  return result;
}

async function main() {
  const text = await input('Enter text to encode/decode: ');

  rl.close();

  text.toLowerCase();

  const result = atbashCipher(text);

  console.log(`Result: ${result}`);
}

main();